test/test_fsdp.py

```diff
@@ -97,12 +99,34 @@ def fsdp_main(rank, world_size, args):
     bsz_local_end = int((rank + 1) / world_size * B)
     ref_input_local = ref_input_global[bsz_local_start:bsz_local_end].to(rank)
 
-    for _ in range(N_ITER):
+    # We first run one iteration without compile, as a workaround to compile float8 layer.
+    # In the first iter, float8 layers go to the branches of "self.is_amax_initialized == False"
+    # After that, float8 layers go the the branches of "self.is_amax_initialized == True"
+    # TODO: Need to fix compile to run wihtout this workaround.
+    optimizer.zero_grad()
+    y_local = model(ref_input_local)
+    y_local.sum().backward()
+    sync_float8_amax_and_scale_history(model)
+    optimizer.step()
+
+    sync_float8_func = sync_float8_amax_and_scale_history
+    if compile:
+        sync_float8_func = torch.compile(
+            sync_float8_amax_and_scale_history, fullgraph=fullgraph
+        )
+        # model = FSDP(torch.compile(model.module), use_orig_params=True)
+        model = torch.compile(model, fullgraph=fullgraph)
```

Comment:
The officially supported way is FSDP(torch.compile(model)). However, we need to run one iteration and then compile the model, which is not very easy if we use FSDP(torch.compile(model)).

Currently, this part worked and could pass the numerics check.